### PR TITLE
feat: Add non-4626 related Ethena functions (SC-839)

### PR DESCRIPTION
**Pull request by lucas-manuel**

### CODE REVIEW

File: test/mainnet-fork/Ethena.t.sol [hunk 1]
```diff
@@ -95,10 +95,38 @@ contract MainnetControllerPrepareUSDeMintFailureTests is ForkTestBase {
         mainnetController.prepareUSDeMint(100);
     }
 
+    function test_prepareUSDeMint_rateLimitBoundary() external {
+        vm.startPrank(SPARK_PROXY);
+        rateLimits.setRateLimitData(
+            mainnetController.LIMIT_USDE_MINT(),
+            100e18,
```
Comment (hexonaut) [inline test/mainnet-fork/Ethena.t.sol:102]: Use realistic numbers. This is USDC so should be `e6` size.

File: test/mainnet-fork/Ethena.t.sol [hunk 2]
```diff
@@ -95,10 +95,38 @@ contract MainnetControllerPrepareUSDeMintFailureTests is ForkTestBase {
...
 
+    bytes32 key;
+
+    function setUp() public override {
+        super.setUp();
+
+        key = mainnetController.LIMIT_USDE_MINT();
+
+        vm.prank(SPARK_PROXY);
+        rateLimits.setRateLimitData(key, 100e18, uint256(10e18) / 1 hours);
```
Comment (hexonaut) [inline test/mainnet-fork/Ethena.t.sol:127]: Again use USDC decimals of 6.

**Review by hexonaut — CHANGES_REQUESTED** [review]

File: test/mainnet-fork/Ethena.t.sol (hunk 2 above)
Comment (lucas-manuel) [reply test/mainnet-fork/Ethena.t.sol]: Updated

File: test/mainnet-fork/Ethena.t.sol (hunk 1 above)
Comment (lucas-manuel) [reply test/mainnet-fork/Ethena.t.sol]: Updated

File: test/mainnet-fork/Ethena.t.sol [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+contract MainnetControllerRemoveDelegatedSignerSuccessTests is ForkTestBase {
+
+    event DelegatedSignerRemoved(address indexed removedSigner, address indexed initiatedBy);
+
+    function test_removeDelegatedSigner() external {
+        address signer = makeAddr("signer");
+        vm.prank(relayer);
+        vm.expectEmit(ETHENA_MINTER);
+        emit DelegatedSignerRemoved(signer, address(almProxy));
+        mainnetController.removeDelegatedSigner(signer);
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/mainnet-fork/Ethena.t.sol:92]: Updated

File: test/mainnet-fork/Ethena.t.sol [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+contract MainnetControllerSetDelegatedSignerSuccessTests is ForkTestBase {
+
+    event DelegatedSignerInitiated(address indexed delegateTo, address indexed initiatedBy);
+
+    function test_setDelegatedSigner() external {
+        address signer = makeAddr("signer");
+        vm.prank(relayer);
+        vm.expectEmit(ETHENA_MINTER);
+        emit DelegatedSignerInitiated(signer, address(almProxy));
+        mainnetController.setDelegatedSigner(signer);
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/mainnet-fork/Ethena.t.sol:46]: Updated

File: test/mainnet-fork/ForkTestBase.t.sol [hunk 5]
```diff
@@ -43,6 +44,14 @@ interface IBufferLike {
     function approve(address, address, uint256) external;
 }
 
+interface ISUSDELike is IERC4626 {
+    function asset() external view returns(address);
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply test/mainnet-fork/ForkTestBase.t.sol]: Removed

File: src/MainnetController.sol [hunk 6]
```diff
@@ -18,10 +20,22 @@ interface IDaiUsdsLike {
...
+    function setDelegatedSigner(address delegateSigner) external;
+    function removeDelegatedSigner(address delegateSigner) external;
+}
+
 interface ISUSDSLike is IERC4626 {
     function usds() external view returns(address);
 }
 
+interface ISUSDELike is IERC4626 {
+    function asset() external view returns(address);
```
[the remark this answers is not in the document]
Comment (lucas-manuel) [reply src/MainnetController.sol]: Removed

**Review by hexonaut — APPROVED** [review]

**Pull request merged**